vintasoftware/drf-rw-serializers
# Update `get_serializer_class` to consider the request method

**Description:** Describe in a couple of sentence what this PR adds

- Update `get_serializer_class` to consider the request method. See the updated docstring.
   - If the request method is GET, it tries to use `self.read_serializer_class`.
   - If the request method is not GET, it tries to use `self.write_serializer_class`.
   - If the specific serializer class for the request method is not set, it falls back to `self.serializer_class`.


> [!NOTE]  
> Once this PR is approved and merged, I will open another one updating the package version so we can publish it to PyPI.

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

Resolves #1
Resolves #17
Resolves #44

**Merge deadline:** List merge deadline (if any)

N/A

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

- On a local testing project, get the changes made, and access the API in the browser:
   - Make sure the page renders properly.
   - Make sure GET and non-GET requests (such as POST) work properly. 
   - Configure a swagger and make sure the UI works properly.
   - Test views using a mix of `serializer_class`, `read_serializer_class`, and `write_serializer_class`.

[Screencast from 23-05-2024 16:55:32.webm](https://github.com/vintasoftware/drf-rw-serializers/assets/22326737/7d7a4e45-c3cf-43c1-82b3-8836246c6682)


**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
